encord/objects/ontology_labels_impl.py

```diff
@@ -1514,17 +1514,43 @@ def to_encord_dict(self) -> dict:
 
     def workflow_reopen(self) -> None:
         """
-        A label row is returned to the first annotation stage for re-labeling. No data is lost during the call.
+        A label row is returned to the first annotation stage for re-labeling.
         No data will be lost during this call.
-        This method is only relevant for the projects that use the Encord Task Management System 2,
-        and does nothing for other types of projects
+
+        This method is only relevant for the projects that use the :ref:`Workflow <tutorials/workflows:Workflows>`
+        feature, and will raise an error for pre-workflow projects.
+
+        Raises:
+            WrongProjectTypeError: when called for pre-workflow project
         """
         if self.label_hash is None:
             # Label has not yet moved from the initial state, nothing to do
             return
 
         self._project_client.workflow_reopen([self.label_hash])
 
+    def workflow_complete(self) -> None:
+        """
+        A label row is returned to the final workflow node.
+
+        This method can be called only for labels for which `.initialise_labels()` was called ance.
+        It is not required to do that every time though.
+
+        This method is only relevant for the projects that use the :ref:`Workflow <tutorials/workflows:Workflows>`
+        feature, and will raise an error for pre-workflow projects
+
+        Raises:
```

Comment:
Or also networking errors for that matter.

Reply:
Totally agree, exception documentation is always like that, as in any case we can divide by zero or do something else stupid inside our code. 
On another side, copy-pasting all possible exceptions everywhere might overbloat our comments very quickly.

We might skip the exception list and just make more detailed comments in free form to not to create an impression that this code can throw _only_ these exceptions. 
I doubt anyone will ever read this if the description will become longer than couple sentences though, so need to think how to phrase that :D

Reply:
Yeah, as I wrote above, totally fine to not to specify exceptions, but right now it is mostly used to explain customer what to do when they see the exception, rather than to list them. 
E.g. here:
```
LabelRowError: when called for a label row fir which labelling has never been initialised
```
it is not to tell them that LabelRowError can be thrown, but mostly to document why it has been thrown and what to do when they see it (initialise labels). 

But we can skip it as long as exception messages are always elaborate enough.

Reply:
okay that is a super valid use case then!